src/snowcli/cli/nativeapp/commands.py

```diff
@@ -59,3 +64,25 @@ def nativeapp_bundle(
 
     except Exception as e:
         return OutputData.from_string(str(e)).add_exit_code(1)
+
+
+@app.command("diff")
+@with_output
+@global_options_with_connection
+def stage_list(
+    stage_fqn: str = typer.Argument(None, help="Name of stage"),
+    folder_name: str = typer.Argument(None, help="Path to local folder"),
+    **options,
+) -> OutputData:
+    """
+    Diffs a stage with a local folder
+    """
+
+    diff = stage_diff(Path(folder_name), stage_fqn)
```

Comment:
nit: Can we make the type explicit since diff is also being used as the command name here?
```suggestion
    diff : DiffResult = stage_diff(Path(folder_name), stage_fqn)
```
with the import.